YogaKit/Source/YGLayout.m

```diff
@@ -304,10 +306,15 @@ static YGSize YGMeasureView(
   const CGFloat constrainedHeight = (heightMode == YGMeasureModeUndefined) ? CGFLOAT_MAX: height;
 
   UIView *view = (__bridge UIView*) YGNodeGetContext(node);
-  const CGSize sizeThatFits = [view sizeThatFits:(CGSize) {
-    .width = constrainedWidth,
-    .height = constrainedHeight,
-  }];
+  CGSize sizeThatFits = CGSizeZero;
+
+  // Fix for https://github.com/facebook/yoga/issues/606
+  if (!view.yoga.isUIView || [view.subviews count] > 0) {
```

Comment:
I think `!view.yoga.isUIView` this check is not required. Happy to hear about your inputs

Reply:
this condition practically says "we want to invoke `sizeThatFits` only in case if the measured view is not a `UIView` instance or it's a `UIView`, but has got children". The check is needed because we do want to invoke `sizeThatFits` for a UILabel (for example) that has got no subviews, but returns a correct value as a result of invocation of `sizeThatFits`

Reply:
Aah Got it. Can you add a comment(in brief) explaining this before the condition?

Reply:
Done